lib/semian/net_http.rb

```diff
@@ -87,14 +88,24 @@ def disabled?
       raw_semian_options.nil?
     end
 
+    def fatal_server_errors?
+      raw_semian_options.fetch(:fatal_server_errors)
+    end
+
     def connect
       return raw_connect if disabled?
       acquire_semian_resource(adapter: :http, scope: :connection) { raw_connect }
     end
 
     def request(req, body = nil, &block)
       return raw_request(req, body, &block) if disabled?
-      acquire_semian_resource(adapter: :http, scope: :query) { raw_request(req, body, &block) }
+      acquire_semian_resource(adapter: :http, scope: :query) do
+        result = raw_request(req, body, &block)
+
+        result.error! if result.is_a?(::Net::HTTPServerError) && fatal_server_errors?
```

Comment:
My point is that if you start raising, you've changed the API for `Net::HTTP`. That is something we've been tempted to do many times in the course of developing Semian (changing client drivers behavior based on failure), but it leaks the abstraction to clients.

I don't think you should raise, I think you should just return the `5xx` quicker. Whether you get a `5xx` directly from the server, or you're hitting a circuit and getting it quicker, the client should deal with it the same way.